File: src/main/java/org/acg/user_services/service/HouseKeepingServiceImpl.java
package org.acg.user_services.service;

import org.acg.user_services.repository.PasswordResetTokenRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

import java.sql.Date;

@Service
public class HouseKeepingServiceImpl implements HouseKeepingService {

    @Autowired
    private PasswordResetTokenRepo passwordResetTokenRepo;

    @Scheduled(cron = "0, 0, *, *, *, *") // Everyday @ Midnight
    @Override
    public void housekeep() {
        passwordResetTokenRepo.deleteAllByExpireDateBefore(new Date(System.currentTimeMillis()));
    }
}

File: src/main/kotlin/org/acg/user/service/HouseKeepingServiceImpl.kt
package org.acg.user.service;

import org.acg.user.repository.PasswordResetTokenRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

import java.sql.Date;

@Service
class HouseKeepingServiceImpl : HouseKeepingService {

    @Autowired
    private lateinit var passwordResetTokenRepo: PasswordResetTokenRepo

    @Scheduled(cron = "0, 0, *, *, *, *") // Everyday @ Midnight
    override fun housekeep() {
        passwordResetTokenRepo.deleteAllByExpireDateBefore( Date(System.currentTimeMillis()))
    }
}
